apple/internal/platform_support.bzl

```diff
@@ -88,13 +96,26 @@ def _platform_prerequisites(
       platform_type_string: The platform type for the current target as a string.
       uses_swift: Boolean value to indicate if this target uses Swift.
       xcode_version_config: The `apple_common.XcodeVersionConfig` provider from the current context.
+      environment: "device" or "simulator" environment of the current target. Optional.
 
     Returns:
       A struct representing the collected platform information.
     """
     platform_type_attr = getattr(apple_common.platform_type, platform_type_string)
     platform = apple_fragment.multi_arch_platform(platform_type_attr)
```

Comment:
[review-note]
Here we don't have an option to get correct platform for simulator as [platform_type](https://bazel.build/rules/lib/toplevel/apple_common#platform_type) is can be only ios, tvos but [platform](https://bazel.build/rules/lib/toplevel/apple_common#platform) can be  ios_simulator as well as ios_device. Same for tvos_device and tvos_simulator.